Modules/fcntlmodule.c

```diff
@@ -246,17 +250,46 @@ fcntl_ioctl_impl(PyObject *module, int fd, unsigned int code,
         }
 
         PyErr_Clear();
-        if (!PyArg_Parse(ob_arg,
-             "i;ioctl requires a file or file descriptor,"
-             " an integer and optionally an integer or buffer argument",
-             &arg)) {
-          return NULL;
+        if (PyLong_Check(ob_arg)) {
```

Comment:
My guess is that the `i` code allowed integer-compatible objects, while you are mandating an actual int here. This could be a regression. Perhaps use PyNumber_Index()?